File: packages/design-tokens-flutter/lib/src/typography.dart
import 'package:flutter/painting.dart';
import 'colors.dart';

/// Source of truth: docs/ui/01_StudentPortal_Web/00_design-system.css.
/// `Outfit` is the canonical UI font; falls back to system + Roboto.
/// `Space Mono` for code / numerics.
class AlpFontFamily {
  static const String ui = 'Outfit';
  static const String mono = 'SpaceMono';
}

class AlpTextStyles {
  AlpTextStyles._();

  static const TextStyle pageTitle = TextStyle(
    fontFamily: AlpFontFamily.ui,
    fontSize: 20,
    fontWeight: FontWeight.w600,
    color: AlpColors.textPrimary,
  );

  static const TextStyle sectionHeading = TextStyle(
    fontFamily: AlpFontFamily.ui,
    fontSize: 16,
    fontWeight: FontWeight.w600,
    color: AlpColors.textPrimary,
  );

  static const TextStyle subheading = TextStyle(
    fontFamily: AlpFontFamily.ui,
    fontSize: 14,
    fontWeight: FontWeight.w600,
    color: AlpColors.textPrimary,
  );

  // Base body is 13px in design-system.css (was 14 in legacy mobile theme).
  static const TextStyle body = TextStyle(
    fontFamily: AlpFontFamily.ui,
    fontSize: 13,
    fontWeight: FontWeight.w400,
    color: AlpColors.textSecondary,
  );

  static const TextStyle label = TextStyle(
    fontFamily: AlpFontFamily.ui,
    fontSize: 12,
    fontWeight: FontWeight.w500,
    color: AlpColors.textSecondary,
  );

  static const TextStyle hint = TextStyle(
    fontFamily: AlpFontFamily.ui,
    fontSize: 12,
    fontWeight: FontWeight.w400,
    color: AlpColors.textMuted,
  );

  static const TextStyle badge = TextStyle(
    fontFamily: AlpFontFamily.ui,
    fontSize: 11,
    fontWeight: FontWeight.w500,
  );

  static const TextStyle micro = TextStyle(
    fontFamily: AlpFontFamily.ui,
    fontSize: 11,
    fontWeight: FontWeight.w400,
    color: AlpColors.textMuted,
  );

  // AI-feature styling — paired with the ◈ glyph at every callsite.
  static const TextStyle aiAccent = TextStyle(
    fontFamily: AlpFontFamily.ui,
    fontSize: 13,
    fontWeight: FontWeight.w500,
    color: AlpColors.colorAi,
  );

  static const TextStyle buttonSm = TextStyle(
    fontFamily: AlpFontFamily.ui,
    fontSize: 12,
    fontWeight: FontWeight.w500,
  );
  static const TextStyle buttonMd = TextStyle(
    fontFamily: AlpFontFamily.ui,
    fontSize: 13,
    fontWeight: FontWeight.w500,
  );
  static const TextStyle buttonLg = TextStyle(
    fontFamily: AlpFontFamily.ui,
    fontSize: 14,
    fontWeight: FontWeight.w500,
  );
}

File: packages/design-system/src/tokens/typography.ts
// Typeface + scale per docs/ui/01_StudentPortal_Web/00_design-system.css.
// Outfit is the canonical UI font; Space Mono for code/numerics.

export const typography = {
  family: {
    ui: '"Outfit", system-ui, -apple-system, "Segoe UI", Roboto, sans-serif',
    mono: '"Space Mono", "Fira Code", "JetBrains Mono", monospace',
  },
  scale: {
    pageTitle: { size: 20, weight: 600 },
    sectionHeading: { size: 16, weight: 600 },
    subheading: { size: 14, weight: 600 },
    body: { size: 13, weight: 400 }, // 13px base per design-system.css
    label: { size: 12, weight: 500 },
    hint: { size: 12, weight: 400 },
    badge: { size: 11, weight: 500 },
    micro: { size: 11, weight: 400 },
    columnHeader: {
      size: 11,
      weight: 600,
      letterSpacing: "0.05em",
      textTransform: "uppercase" as const,
    },
    button: { sm: 12, md: 13, lg: 14, weight: 500 },
  },
} as const;

export type TypographyTokens = typeof typography;
